5a/SiloHost/src/Program.fs

```diff
@@ -39,20 +39,74 @@ let ipAddress () : Async<IPAddress> =
             }
     }
 
-let advertisedIpAddress () : Async<IPAddress> = raise(NotImplementedException())
+let siloPort () : Async<int> =
+    async {
+        let parsed, siloPort =
+            Environment.GetEnvironmentVariable("SILOPORT")
+            |> Int32.TryParse
+
+        return
+            match parsed with
+            | true -> siloPort
+            | false -> raise (ArgumentException("SILOPORT environment variable not set"))
+    }
 
-let siloPort () : Async<int> = raise(NotImplementedException())
+let gatewayPort () : Async<int> =
+    async {
+        let parsed, gatewayPort =
+            Environment.GetEnvironmentVariable("GATEWAYPORT")
+            |> Int32.TryParse
+
+        return
+            match parsed with
+            | true -> gatewayPort
+            | false -> raise (ArgumentException("GATEWAYPORT environment variable not set"))
+    }
 
-let gatewayPort () : Async<int> = raise(NotImplementedException())
+let primarySiloPort () : Async<int> =
+    async {
+        let parsed, primarySiloPort =
+            Environment.GetEnvironmentVariable("PRIMARYPORT")
+            |> Int32.TryParse
 
-let primarySiloPort () : Async<int> = raise(NotImplementedException())
+        return
+            match parsed with
+            | true -> primarySiloPort
+            | false -> raise (ArgumentException("PRIMARYPORT environment variable not set"))
+    }
 
-let dashboardPort() : Async<int> = raise(NotImplementedException())
+let dashboardPort () : Async<int> =
+    async {
+        let parsed, dashboardPort =
+            Environment.GetEnvironmentVariable("DASHBOARDPORT")
+            |> Int32.TryParse
 
-let primarySiloEndpoint() : Async<IPEndPoint> = raise(NotImplementedException())
+        return
+            match parsed with
+            | true -> dashboardPort
+            | false -> raise (ArgumentException("DASHBOARDPORT environment variable not set"))
+    }
 
 [<EntryPoint>]
 let main args =
+    let task = async {
```

Comment:
No sure the variable name is most fortunate:

* it's singular
* the object is not really a task, but an expression

Reply:
https://github.com/PiotrJustyna/road-to-orleans/pull/305/commits/207c099170a0fd15baf00b4d1d38336124754cc8